### PR TITLE
SERV-492 Review

### PR DESCRIPTION
**Pull request by ksclarke**
* Update dependency versions to fix security issues
* Remove remaining native-image stuff since the staged build was removed from Dockerfile
* Remove testcontainers stuff (assuming we'll just use docker-maven-plugin(?))
* Remove "security fix" dependencies since the version upgrade also fixes them
* Update alpine image version
* Update alpine image to use JDK 17
* Add the UCLA Library Snyk Org ENV

Questions:
* Do we want to support Maven wrapper (the new Checkstyle suppressions adds support for that)? We added it to one recent project, but I'm not clear if that's something we want to continue to do(?)

Note:
* I didn't bump parent project version because I seem to have hosed the latest version of the parent project's logging setup :-(

### CODE REVIEW

File: src/main/docker/Dockerfile [hunk 1]
```diff
@@ -10,7 +10,7 @@ LABEL ContainerName="${project.name}" ContainerSourceCode=${project.url}
 
 # Install packages and setup the application's user
 WORKDIR /opt/${project.artifactId}
-RUN apk add openjdk11 --quiet --no-cache \
+RUN apk add openjdk17 --quiet --no-cache \
```
Comment (markmatney) [inline src/main/docker/Dockerfile:13]: What about the other places where Java 11 is referenced (maven.compiler.release, etc.)?

File: src/main/tools/checkstyle/checkstyle-suppressions.xml [hunk 2]
```diff
@@ -6,4 +6,5 @@
 <suppressions>
   <suppress files="src[\\/]main[\\/]generated[\\/]" checks="."/>
   <suppress files="target[\\/]" checks="."/>
+  <suppress files=".mvn[\\/]" checks="."/>
```
Comment (markmatney) [inline src/main/tools/checkstyle/checkstyle-suppressions.xml:9]: I won't personally be using mvnw, but I have no preference on whether to add it or not

Comment (ksclarke) [reply src/main/tools/checkstyle/checkstyle-suppressions.xml:9]: Let's not then. I'm not a huge fan of it.

File: src/main/docker/Dockerfile (hunk 1 above)
Comment (ksclarke) [reply src/main/docker/Dockerfile:13]: The parent project sets that dynamically depending on which JDK it finds on the system.

File: src/main/tools/checkstyle/checkstyle-suppressions.xml (hunk 2 above)
Comment (ksclarke) [reply src/main/tools/checkstyle/checkstyle-suppressions.xml:9]: It doesn't hurt to use this suppressions file as is though.

File: src/main/docker/Dockerfile (hunk 1 above)
Comment (markmatney) [reply src/main/docker/Dockerfile:13]: I see, nevermind then, I was not aware of that

**Review by markmatney — APPROVED** [review]

**Pull request merged**